### ISSUE TITLE
How to deploy cloudcore and edgecore on one host？

### ISSUE THREAD

**Issue by CX1900** [kubeedge/kubeedge#2553]
**What happened and what you expected to happen**:
I used keadm to deploy both cloudcore and edgecore on one host, after I used keadm join command to deploy edgecore, logs said edgecore is running, but I can't find the node of the edgecore by using command kubectl get node.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
<details><summary>Kubernets Version</summary>

```console
$ kubectl --version
# paste output here
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}

</details>

<details><summary>CloudCore Version</summary>

```console
$ cloudcore --version
# paste output here
```
kubeedge-v1.5.0
</details>

<details><summary>EdgeCore Version</summary>

```
$ edgecore --version
# paste output here
```
kubeedge-v1.5.0
</details>

> linux version
Linux version 3.10.0-1160.6.1.el7.x86_64 (mockbuild@kbuilder.bsys.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-44) (GCC) ) #1 SMP Tue Nov 17 13:59:11 UTC 2020

**Comment by daixiang0** [comment]
Please double check edgecore process exists or not first.

**Comment by CX1900** [comment]
> Please double check edgecore process exists or not first.

It's strange that when I used keadm join command to deploy edgenode on another host, although I can't find edgecore process on the edge host, I can find edgenode showing ready by using kubectl get node, counter-app demo also worked. While I deployed edgecore on the same host with cloudcore, I can't find edgenode by using kubectl get node as well as edgecore process. Finding that edgecore process didn't exist, I tried to execute keadm join command again, here are the errors it reported:
/*errors report starts*/
Host has mosquit+ already installed and running. Hence skipping the installation steps !!!
**Error: EdgeCore is already running on this node, please run reset to clean up first**

F0203 11:10:34.490022    5239 keadm.go:27] **EdgeCore is already running on this node, please run reset to clean up first
goroutine 1 [running]:**
k8s.io/klog/v2.stacks(0xc000010001, 0xc000152000, 0x78, 0xcc)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:996 +0xb8
k8s.io/klog/v2.(*loggingT).output(0x2f84ce0, 0xc000000003, 0x0, 0x0, 0xc0002141c0, 0x2ca1e32, 0x8, 0x1b, 0x1ea1f00)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:945 +0x19d
k8s.io/klog/v2.(*loggingT).printDepth(0x2f84ce0, 0xc000000003, 0x0, 0x0, 0x1, 0xc000da3f68, 0x1, 0x1)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:718 +0x15e
k8s.io/klog/v2.(*loggingT).print(...)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:703
k8s.io/klog/v2.Fatal(...)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1436
main.main()
	/data/gopath/src/github.com/kubeedge/kubeedge/keadm/cmd/keadm/keadm.go:27 +0xa1

goroutine 6 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x2f84ce0)
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1131 +0x8b
created by k8s.io/klog/v2.init.0
	/data/gopath/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:416 +0xd6
/*errors report ends*/
After that, I tried keadm reset on the host supposed to have both cloudcore and edgecore, but only cloudcore was stopped. I'd appriciate it if you could give me some possible directions to solve it.

**Comment by daixiang0** [comment]
`ps faux | grep edgecore` with root user to check it.

Also, put reset log here.

**Comment by CX1900** [comment]
> `ps faux | grep edgecore` with root user to check it.
> 
> Also, put reset log here.

Host supposed to have both cloudcore and edgecore
sudo ps faux | grep edgecore
root      6220  0.0  0.0 112808   968 pts/2    S+   12:04   0:00          \_ grep --color=auto edgecore
root     10199  4.9  0.2 3029052 65524 ?       Ssl  10:42   4:03 /etc/kubeedge/edgecore
Host that only deployed edgecore
sudo ps faux | grep edgecore
centos   30687  0.0  0.0 116940  1004 pts/2    S+   04:04   0:00              \_ grep --color=auto edgecore
root     25822  2.9  0.8 1987340 65552 ?       Ssl  03:23   1:13 /etc/kubeedge/edgecore

**Comment by daixiang0** [comment]
If you want to deploy cloudcore and edgecore at same host, check https://github.com/kubeedge/kubeedge/blob/master/hack/local-up-kubeedge.sh.

On the edge node, you need check this process is controlled by which, nohup in the background or systemd, then kill it.

`keadm reset` should work, if not, put log here.

**Comment by CX1900** [comment]
Recently, I tried to [Deploy kubeedge's edge nodes using containers](https://github.com/kubeedge/kubeedge/tree/master/build/edge), after I executed "./run_daemon.sh up", I found the edge container exited with 255 code inmediately using docker ps -a, any clue about it?

Executing results are as follows:
sudo ./run_daemon.sh up
WARNING: Some services (edgecore) use the 'deploy' key, which will be ignored. Compose does not support 'deploy' configuration - use `docker stack deploy` to deploy to a swarm.
edge_dind_1 is up-to-date
edge_emqx_1 is up-to-date
Starting edge_edgecore_1 ... done
